content/recipes/swagger.md

```diff
@@ -160,6 +160,117 @@ With `isArray` set to **true**, the `enum` can be selected as a **multi-select**
 
 <figure><img src="/assets/enum_query_array.gif" /></figure>
 
+#### Enums Schema
+
+By default, `enum` property will add a raw definition of [Enum](https://swagger.io/docs/specification/data-models/enums/) on the `parameter`.
+
+```yaml
+CatDetail:
+  type: 'object'
+  properties:
+    ...
+    - breed:
+        type: 'string'
+        enum:
+          - Persian
+          - Tabby
+          - Siamese
+    ...
+    
+CatInformation:
+  type: 'object'
+  properties:
+    ...
+    - breed:
+        type: 'string'
+        enum:
+          - Persian
+          - Tabby
+          - Siamese
+    ...
+```
+
+The above specification works fine for most cases. However, if you are utilizing a tool that takes the specification as **input** and generates **client-side** code, you might run into a problem with the generated code containing duplicated `enums`. Check the following snippet out: 
+
+```typescript
+// generated client-side code
+export class CatDetail {
+  ...
+   breed: CatDetailEnum;
+   ...
+}
+
+export class CatInformation {
+  ...
+  breed: CatInformationEnum;
+  ...
+}
+
+export enum CatDetailEnum {
+  Persian = 'Persian',
+  Tabby = 'Tabby',
+  Siamese = 'Siamese'
+}
+
+export enum CatInformationEnum {
+  Persian = 'Persian',
+  Tabby = 'Tabby',
+  Siamese = 'Siamese'
+}
+```
+
+> info **Hint** The above snippet is generated using a tool called [NSwag](https://github.com/RicoSuter/NSwag).
+
+You can see that now you have two `enums` that are exactly the same. 
+To address this issue, you can pass an `enumName` next to `enum` property in your decorator.
+
+```typescript
+export class CatDetail {
+  ...
+   @ApiProperty({ enum: CatBreed, enumName: 'CatBreed' })
+   breed: CatBreed;
+   ...
+}
+
+export class CatInformation {
+  ...
+   @ApiProperty({ enum: CatBreed, enumName: 'CatBreed' })
+   breed: CatBreed;
+   ...
+}
+```
+
+`enumName` allows `nestjs/swagger` to be able to turn `CatBreed` into its own `schema` which in turns makes `CatBreed` reusable. The specification will look like the following:
```

Comment:
`enumName` allows `nestjs/swagger` to be able to turn
->
`enumName` enables `nestjs/swagger` to turn